app/models/presenter/bid.rb

```diff
@@ -9,7 +9,11 @@ def bidder_duns_number
     end
 
     def bidder_name
-      bidder.name || null
+      if Presenter::Auction.new(auction).available?
+        '[Name witheld until the auction ends]'
+      else
+        bidder.name || null
```

Comment:
woah! I don't usually see `null` returned like this in Ruby -- in fact, I don't think it is a thing in Ruby (but definitely correct me if I am wrong!)

```ruby
[1] pry(main)> null
NameError: undefined local variable or method `null' for main:Object
from (pry):1:in `<main>'
[2] pry(main)> nil
=> nil
```

if `bidder.name` is nil, this will just return `nil` -- would that work for your use case?